Controllers/HomeController.cs

```diff
@@ -8,100 +9,124 @@
 using System.Threading.Tasks;
 using TempleTours.Models;
 using TempleTours.Models.Database;
+using TempleTours.Models.ViewModels;
 
 namespace TempleTours.Controllers
 {
     public class HomeController : Controller
     {
         private ITempleTourRepo _repository;
-        private TempleTourDbContext _context;
-        public Tour UnfinishedTour { get; set; }//used as a placeholder
 
         private readonly ILogger<HomeController> _logger;
 
+        private TempleTourDbContext _context;
+
         public HomeController(ILogger<HomeController> logger, ITempleTourRepo repo, TempleTourDbContext context)
         {
             _logger = logger;
             _repository = repo;
             _context = context;
         }
 
-       
-
         public IActionResult Index()
         {
+            _logger.LogInformation("GET request on Index.");
             //this one is ready
             return View();
         }
 
         public IActionResult Privacy()
         {
+            _logger.LogInformation("GET request on Privacy.");
             return View();
         }
+
         [HttpGet]
         public IActionResult SignUp()
         {
-
-
-            
+            _logger.LogInformation("GET request on SignUp.");
             //This one should be ready to go
-            return View(_repository.Tours.OrderBy(t => t.TourTime));
+            return View(from t in _repository.Tours
```

Comment:
This is the other way to write LINQ. Equivalent to _repository.Tours.Where(t => t.Parties.Any()).Where(t => t.TourTime > DateTime.Now).OrderBy(t => t.TourTime).